src/target/parsers/cpu.cc

```diff
@@ -28,7 +28,24 @@ namespace target {
 namespace parsers {
 namespace cpu {
 
+Optional<String> DetectSystemTriple() {
+  auto pf = tvm::runtime::Registry::Get("target.llvm_get_system_triple");
+  if (pf->defined()) {
+    return (*pf)();
+  }
+  return {};
+}
+
 TargetJSON ParseTarget(TargetJSON target) {
+  String kind = Downcast<String>(target.Get("kind"));
+  Optional<String> mtriple = Downcast<Optional<String>>(target.Get("mtriple"));
+  Optional<String> mcpu = Downcast<Optional<String>>(target.Get("mcpu"));
+
+  // Try to fill in the blanks by detecting target information from the system
+  if (kind == "llvm" && !mtriple.defined() && !mcpu.defined()) {
+    target.Set("mtriple", DetectSystemTriple().value_or(""));
+  }
```

Comment:
One small nit, in this case would be nice to leave a message for the user about facts:

 * explicit mtriple and/or explict mcpu wasnt provided
 * backend here detected an X mtriple and/or and Y mcpu

Reply:
Just added, thanks!

Reply:
After looking through the CI logs, I'm wondering if emitting a warning message is not the best idea here - there are way too many warnings. Perhaps it can be implemented in some driver of tvm instead, e.g. TVMC, and emitted once?